### PR TITLE
Mac python support

### PR DESCRIPTION
**Pull request by jeppojeps**
Added automatic support for MacOS that has homebrew installed and can already build with make, cmake, clang

### CODE REVIEW

File: shuriken/bindings/Python/setup.py [hunk 1]
```diff
@@ -56,6 +56,27 @@ def build_libraries():
     else:
         logger.warning("CMake build is only supported on Linux for now.")
 
+    if platform.system() == 'Darwin':
+        try:
+            with change_directory(BUILD_FOLDER):
```
Comment (badumbatish) [inline shuriken/bindings/Python/setup.py:61]: i think if the code is identical we should move them to the same if condition.

Also here we're letting the logger leak out a warning of linux being supported even if we're compiling on Darwin

**Review by badumbatish — CHANGES_REQUESTED** [review]

**Comment by badumbatish** [comment]
once this merges I'll get started on #107

File: shuriken/bindings/Python/setup.py (hunk 1 above)
Comment (jeppojeps) [reply shuriken/bindings/Python/setup.py:41]: Morning I just pushed the change, we will need to keep an eye on the setup.py script as MacOS may change <sigh> abruptly

Comment (badumbatish) [reply shuriken/bindings/Python/setup.py:41]: Yes i agree, good work! :)

**Comment by badumbatish** [comment]
Can this be squashed into 1 commit?

**Pull request merged**